### PR TITLE
BaSP-TG-74: add and edit projects using redux

### PR DESCRIPTION
**Pull request by AlejoFGM**

### CODE REVIEW

**Review by GinaSchP — APPROVED** [review]

File: src/Components/Projects/Form/index.js [hunk 1]
```diff
@@ -60,21 +57,21 @@ const Form = () => {
           }
           return 'noEmployee';
         });
-        const newEmployeeList = employeeList.filter((employee) => employee !== 'noEmployee');
+        const newEmployeeList = employeeList?.filter((employee) => employee !== 'noEmployee');
```
Comment (lmalvasia) [inline src/Components/Projects/Form/index.js:60]: The `?` should not be necessary.

File: src/Components/Projects/Form/index.js [hunk 2]
```diff
@@ -60,21 +57,21 @@ const Form = () => {
...
         setProjectState({
-          name: project.data.name,
-          clientName: project.data.clientName,
-          description: project.data.description,
-          startDate: project.data.startDate,
-          endDate: project.data.endDate,
+          name: currentProject?.name,
```
Comment (lmalvasia) [inline src/Components/Projects/Form/index.js:62]: The `?` should not be necessary.

File: src/Components/Projects/Form/index.js [hunk 3]
```diff
@@ -85,48 +82,22 @@ const Form = () => {
...
-        const data = await response.json();
-        if (!data.error) {
-          history.push('/projects');
-        } else {
-          setErrorMessage(data.message);
-        }
-      } catch (error) {
-        alert(error.message);
+      const response = await dispatch(postProject(projectState));
+      if (response?.type === POST_PROJECTS_SUCCESS) {
```
Comment (lmalvasia) [inline src/Components/Projects/Form/index.js:86]: The `?` should not be necessary.

File: src/Components/Projects/Form/index.js [hunk 4]
```diff
@@ -16,25 +20,22 @@ const Form = () => {
...
-    try {
-      fetch(`${process.env.REACT_APP_API_URL}/employees`)
-        .then((response) => response.json())
-        .then((response) => {
-          setEmployeeList(response.data);
-        });
-    } catch (error) {
-      alert(error);
-    }
+    dispatch(getEmployees());
```
Comment (lmalvasia) [inline src/Components/Projects/Form/index.js:37]: If we are fetching data, we should show a spinner until the data is fetched.

**Review by lmalvasia — APPROVED** [review]

**Comment by eyrent** [comment]
### QA Report: PASS

**Functionality tested:**
- Create a project with valid data: **PASS** :heavy_check_mark: 
- Edit an existing project with valid data: **PASS** :heavy_check_mark:
- Error messages,relay server messages to user: **PASS** :heavy_check_mark:

### Observations
- Creating or editing a project with no employees assigned is allowed. :yellow_circle: 
- Error messages are not very user friendly. Some even repeat the word Error. These come directly from server response. [1] :yellow_circle:  
- Error messages style could be improved (use red for example) 
- 'Loading' message is not centered. [2] :green_circle: :
- When assigning an employee (with the 'Assign' button) no check is in place to prevent employee with no rate from being added. However the server rejects this and throws an error. [3] :green_circle: 
- The date selector on my set up fails sometimes (Firefox 107, Fedora Linux 36). :green_circle: 

### Suggestions
- Have the team work on server error messages and that would solve some of the observations made.
- Add more validations to server for projects with no employees.
- Add a little styling to error messages. 

### Screenshots
[1]
Error messages are not user firendly
[2]
Loading message is off centered
[3]
Assigning an employee with no rate
Server rejects project with employees without rate

**Pull request merged**